### ISSUE TITLE
Patch `dotnet symbol-tool` to check completeness & availability of symbols

### ISSUE THREAD

**Issue by JohnTortugo** [dotnet/arcade#2499]
After some review / discussions about these issues: #167, #173, #1874 it was clear that issue #1874 encompasses both #167 and #173. Moreover, it was decided that we would use [symbol-tool](https://github.com/dotnet/symstore/tree/master/src/dotnet-symbol) to perform validation and availability check of symbols. 

A temporary solution for both of these problems was provided in this PR: https://github.com/dotnet/arcade/pull/2492 . However, there are a few things that the script in that PR intentionally doesn't do and that we should make sure `symbol tool` does. Moreover, there are a few things we should implement in `symbol tool` as well:

- Receive the path to two local NuGet packages and perform symbol completeness check.
- Receive the path to a regular NuGet package and check if symbols for that package are available in **both** MSDL & SymWeb.
- No need for downloading files when performing availability. Just checking the symbol file is available is probably sufficient.
- The tool should be able to handle "extension-less" files, like Linux executables.
- Make the tool available as an MSBuild task/target ?

For the record:

- **Symbol availability:** check that for every module in a .nupkg there is a symbol file in MSDL ~or~*and* SymWeb.
- **Symbol completeness:** given a .nupkg and a .symbols.nupkg (or .snupkg) we need to check that for each module in the .nupkg there is a symbol file in the symbol package.

/cc @mikem8361 @dagood @tmat @leculver

**Comment by dagood** [comment]
> * **Symbol completeness:** given a .nupkg and a .symbols.nupkg (or .snupkg) we need to check that for each module in the .nupkg there is a symbol file in the symbol package.

This issue might be written shorthand and this is already known, but completeness isn't limited to modules, and most files have more than one associated file. For each file in the nupkg that requires some associated files to be indexed, *all* associated files should exist. I documented the associations I know of at https://github.com/dotnet/arcade/issues/167#issuecomment-478780348.

(The temporary solution doesn't handle these.)

**Comment by dagood** [comment]
This issue came up again recently and I noticed this:

> 
> * Receive the path to a regular NuGet package and check if symbols for that package are available in **both** MSDL & SymWeb.
> ...
> * **Symbol availability:** check that for every module in a .nupkg there is a symbol file in MSDL or SymWeb.

I want to point out that the first is correct, the symbols must be available in MSDL **and** symweb.

**Comment by riarenas** [comment]
We are moving symbol validation to be part of the ring promotion activities. Moving to the appropriate tracking epic

**Comment by adiaaida** [comment]
Out of scope for #6418.

**Comment by jakubstilec** [comment]
[Async Triage]: @adiaaida the same as #2519.

**Comment by markwilkie** [comment]
@mmitche  - do you know if this is still an issue?

**Comment by mmitche** [comment]
I don't think so? It's the standard tool, not one that we own, and the verification appears to be complete (spent some time looking at it yesterday)

**Comment by markwilkie** [comment]
thanks - closing

**Comment by dagood** [comment]
> appears to be complete

Can you be more specific? I don't see anything related in https://github.com/dotnet/symstore/commits/main/src/dotnet-symbol, glancing over it. I would expect significant work there (or somewhere else, where?) to get anywhere close to being able to close this issue.

**IMO someone should go through each bullet point and show how the automated system gets it done before closing this.** Otherwise, we're at risk of having a flaky release pipeline without tracking already known flakiness and known fixes.

I'm highly skeptical of attempts to close this issue and other symbol validation issues because the problem has incorrectly been assumed to be solved multiple times in the past.

/cc @leecow

**Comment by markwilkie** [comment]
reopening because @dagood scared me.   :)

**Comment by dagood** [comment]
In the end, it *could* be closed for project scope reasons... but I think it's important to be clear about what is/isn't being validated so the release team, product teams, and highly impacted users (like Azure's snapshot debugging) can be aware and potentially push for more. What I'm trying to avoid is a surprise. 😄

**Comment by leculver** [comment]
I agree with @dagood.  I would like to see a more systematic review/validation of what's being archived long term.  I'm happy to help provide code that does validation of symbols.

I'm hoping that the Portable PDB work will (eventually) be finished this year, which will remove our requirement to archive Windows PDBs in favor of Portable PDBs only.

Might I suggest postponing this work (but not losing track of it!) until portable PDBs are supported by the dbgeng toolchain?  (ETA later this year.)  At that point making a push to ensure we are archiving the right kinds of symbols in the right places would be both timely and high value.

**Comment by MattGal** [comment]
[Async Triage]: Seems like it should be done, but is it in this repo?

**Comment by markwilkie** [comment]
Moving to the release epic so the v-team can dig in further.
